[REST API] Show JP banner and simplified JP dialog for REST API logins

hafizrahman opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #8595
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

For REST API logins, this PR makes the Jetpack bottom banner on My Store screen shows up. When opened, the screen shows the benefits of installing Jetpack (unlocking various features in the app), and adds a button to allow Jetpack install thru a browser.

This is meant to inform users further that some features require Jetpack, and gives them the option to do so. In the future iteration, the installation process will be native instead of thru browser, so this is a temporary solution.


### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

1. Prepare a .org test site that has no Jetpack plugin installed,
2. Start app, enter the test site's address, then credentials,
3. After being logged in, make sure the Jetpack banner is shown at the bottom of My Store screen.
4. Tap the banner, make sure the Jetpack benefits dialog is shown with two buttons on it ("Open wp-admin" and "Not now")

Test the "Open wp-admin" button
1. Tap the button, ensure a Chrome Custom Tab opens with a Jetpack setup screen loaded,
2. (optional) follow through the setup process until it Jetpack is installed and connected, the browser will then return to the app and it loads the WPCom login screen.

Test the "Not now" button
1. Tap the button, ensure that the dialog is closed.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

| Banner is shown | Dialog | Chrome Custom Tab |
|-|-|-|
| ![jp-banner](https://user-images.githubusercontent.com/266376/227120934-b3fa1cc9-3db3-4869-be99-0997b081f157.png) | ![jp-dialog](https://user-images.githubusercontent.com/266376/227120931-1c2a6b16-1f69-4550-a8e7-ec067f69ddfc.png) | ![jp-cct-install](https://user-images.githubusercontent.com/266376/227120915-ec88a6b0-43cb-4c07-9719-6ee2d461b9e5.png) |


- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hafizrahman committed:
Add: Show Jetpack banner and dialog for REST API login.

The dialog should not have a CTA except to dismiss it.

hafizrahman committed:
Fix: typo.

hafizrahman committed:
Fix: detekt.

pmusolino commented:
Please, make sure to [add an event](https://github.com/woocommerce/woocommerce-ios/pull/9232) to track how people are attempting to install Jetpack from here. @itsmeichigo already added it in her [PR](https://github.com/woocommerce/woocommerce-ios/pull/9232).

hafizrahman committed:
Refactor dismiss button to always appear in all cases.

hafizrahman committed:
Add: functionality to open Jetpack install screen via wp-admin in browser.

hafizrahman committed:
Add: Hook up open Jetpack install in wp-admin functionality to  button in JP benefits dialog.

hafizrahman committed:
Add: Tracking for JP install in wp-admin.

hafizrahman committed:
Add: Release note.

hafizrahman commented:
@pmusolino Thanks! Added in https://github.com/woocommerce/woocommerce-android/pull/8596/commits/f8cf3367691ec5f23c3ec992ad729c6302f76e3d

hafizrahman commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/jetpack/benefits/JetpackBenefitsViewModel.kt:
```diff
@@ -24,10 +26,16 @@ class JetpackBenefitsViewModel @Inject constructor(
             isUsingJetpackCP = selectedSite.connectionType == SiteConnectionType.JetpackConnectionPackage,
-            isLoadingDialogShown = false
+            isLoadingDialogShown = false,
+            isRestAPIActivationAvailable = !FeatureFlag.REST_API_I2.isEnabled()
```
In case this is confusing: Jetpack Activation REST API will be available once the feature flag is removed.

JorgeMucientes replied:
Sorry, I'm still confused 😅 Why do we need this? With the change added [here](https://github.com/woocommerce/woocommerce-android/pull/8596/files#diff-55d57a81402b1912306c79a73967a03d82fe53218b5502e310855b21eaf46dfbL347)  doesn't depend on the feature flag anymore. So, it will be enabled. And this line will show the following buttons in production:

![Screenshot 2023-03-23 at 18 47 58](https://user-images.githubusercontent.com/2663464/227297364-68a6dd65-eada-4be2-b578-8c0f0c70b6e6.png)

and for `debug` builds these:

![Screenshot 2023-03-23 at 18 48 14](https://user-images.githubusercontent.com/2663464/227297499-9eecbcaf-e6e9-43a9-9fcc-88a3203ad7b7.png)


Is that what we want? Why?

JorgeMucientes commented:
Nicely done @hafizrahman. The code for `debug` build works as expected. I just left a question as I'm a bit confused with the feature flag usage. Feel free to ignore it and merge if it's not a concerning issue.

hafizrahman committed:
Simplify display cases:

1. Always show the wp-admin install button (for now)
2. Feature flagged: Show native install button.

hafizrahman commented:
@JorgeMucientes You're right that it's confusing, thanks for raising that :)

I simplified things in https://github.com/woocommerce/woocommerce-android/pull/8596/commits/d1a73f30ed23896fcb11b15e20123982155d3558

1. On all cases, show the wp-login installation button
2. Behind `REST_API_I2` feature flag, show also the native installation button.

This way debug will show 3 buttons (shouldn't be a problem), but production will correctly show the wp-login button and dismiss button only.

hafizrahman committed:
Merge branch 'trunk' into issue/8595-rest-api-show-jp-banner-without-cta
